### PR TITLE
feat : Onboarding

### PR DESCRIPTION
**Pull request by arpit01923**
Functionality:
- firstly app ask for your name
- then give you greet and show time
- added weather feature also which dynamically choose your location and display time
- added quote feature

Preview:
image
image

### CODE REVIEW

**Review by manojncsoni007 — APPROVED** [review]
Nice UI & clean code

File: src/components/weather/weather.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    };
+
+    const error = () => {
+      console.log("Sorry there is an error");
+    };
+    navigator.geolocation.getCurrentPosition(success, error, options);
+  }, []);
+
+  useEffect(() => {
+    getWeather(loc?.lat, loc?.lon, setWeather);
```
Comment (TejasShekar) [inline src/components/weather/weather.jsx:32]: What is the optional parameter being used for ?

File: src/pages/onboarding/onboarding.jsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import { Weather } from "../../components";
+
+export const OnBoarding = () => {
+  const [userName, setUserName] = useState("");
+  const [printUserName, setPrintUserName] = useState("");
+  const [mainFocus, setMainFocus] = useState("");
+  const [printMainFocus, setPrintMainFocus] = useState("");
+  const [hours, setHours] = useState("");
+  const [minutes, setMinutes] = useState("");
+  const [quote, setQuote] = useState("");
```
Comment (TejasShekar) [inline src/pages/onboarding/onboarding.jsx:15]: You can make the time as a seperate component. Makes it modular.

File: src/pages/onboarding/onboarding.jsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import { useEffect, useState } from "react";
+import "./onboarding.css";
+import { AiFillEdit } from "react-icons/ai";
+import { getGreet, getHour, getMinute } from "../../utils";
+import { getQuote, getWeather } from "../../services";
+import { Weather } from "../../components";
+
+export const OnBoarding = () => {
+  const [userName, setUserName] = useState("");
+  const [printUserName, setPrintUserName] = useState("");
```
Comment (TejasShekar) [inline src/pages/onboarding/onboarding.jsx:10]: You can use one state for the same purpose

**Review by TejasShekar — APPROVED** [review]
Good file structure.
You can add a 50% black screen in between the bg-img and the text, so that it provides a good contrast and your text readability improves.
I have added a few suggestions.
Add a preview link if possible from next PR.

File: src/components/weather/weather.jsx (hunk 1 above)
Comment (arpit01923) [reply src/components/weather/weather.jsx:32]: initially value of lat and lon is empty string

File: src/pages/onboarding/onboarding.jsx (hunk 2 above)
Comment (arpit01923) [reply src/pages/onboarding/onboarding.jsx:15]: will do it in future

File: src/pages/onboarding/onboarding.jsx (hunk 3 above)
Comment (arpit01923) [reply src/pages/onboarding/onboarding.jsx:10]: one state is used for username and another state is used  when button is clicked

**Pull request merged**